### PR TITLE
Adding remote resource module

### PR DESCRIPTION
**Pull request by hamedahmadi**

### CODE REVIEW

**Comment by hamedahmadi** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;Y7M4tXdi&#x2F;187-investigate-django-rest-framework-and-json-api-module"> Investigate Django Rest Framework and JSON API module</a>

**Comment by hamedahmadi** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;kvUEh4pB&#x2F;197-the-self-link-for-local-requests-is-broken"> The self link for local requests is broken.</a>

**Comment by hamedahmadi** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;JzVBCkmV&#x2F;196-json-api-finish-up-the-self-link-for-the-collection-view"> JSON API, finish up the &#39;self link&#39; for the collection view.</a>

File: setup.py [hunk 1]
```diff
@@ -10,7 +21,12 @@
       url='https://github.com/ZeroCater/zc_common',
       download_url='https://github.com/ZeroCater/zc_common/tarball/0.0.2',
       license='MIT',
-      packages=['zc_common'],
+      packages=get_packages('zc_common'),
+      install_requires=[
```
Comment (birkholz) [inline setup.py]: Adding django/DRF as required packages means users can't install zc_common just for the timezone utils. Please remove this, and add a note to the README specifying the requirement for this particular part of the package.

File: zc_common/remote_resource/relations.py [hunk 2]
```diff
@@ -9,14 +9,42 @@
...
-        super(RemoteResourceField, self).__init__(*args, **kwargs)
+
+        if related_resource_path is None:
+            raise NameError('related_resource_path parameter must be provided')
+
+        self.related_resource_path = related_resource_path
+
+        super(RemoteResourceField, self).__init__(**kwargs)
+
+    def get_links(self, obj=None, lookup_field='pk'):
```
Comment (ograycode) [inline zc_common/remote_resource/relations.py]: This looks pretty good. I have no opinions on how the links get generated -- we could even make it so every field has to implement it or have it be like default fields in django models (where they take a static value or callable function)

**Pull request merged**